common/src/main/java/io/pravega/common/util/BufferView.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.common.util;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+/**
+ * Defines a generic read-only view of a readable memory buffer with a known length.
+ * For array-backed Buffers, see {@link ArrayView}.
+ */
+public interface BufferView {
```

Comment:
Explain the ref counting and how one is expected to use this class in the javadoc above.

Reply:
I added more explanation there.